### ISSUE TITLE
This method seems only work for classification task

### ISSUE THREAD

**Issue by 61peng** [Shark-NLP/self-adaptive-ICL#4]
For generation task such as _squad_ (which was included in your code), the scores in below always return 0 for all candidate demonstrations because there is just 1 label for generation task, and the selected_idxs always return the first one.

```python
        if self.calibrate and self.task_type != 'QA':
            batch_labels_prior = [[build_instruction(x=query['X'], c=query['C'], e=e,
                                                     y_text=None,
                                                     instruction=self.prompting_instruction[label],
                                                     tokenizer=self.tokenizer,
                                                     e_instruction=self.example_instruction, need_span_ids=self.span,
                                                     max_len=self.n_tokens, prior=True, prior_no=self.prior_no)[0]
                                   for i, e in enumerate(in_context_examples)]
                                  for label in self.labels]  # label:int
            prior_loss_list = []
            for batch in batch_labels_prior:
                with torch.no_grad():
                    prior_ce_loss, prior_lens = evaluate(self, batch, span=self.span)
                    avg_prior_loss = (prior_ce_loss / prior_lens).tolist()
                    prior_loss_list.append(avg_prior_loss)
            scores = get_score(self, batch_labels, method=method, span=self.span, prior_loss_list=prior_loss_list)
        else:
            scores = get_score(self, batch_labels, method=method, span=self.span)
        selected_idxs = all_candidate_idx[scores.argmax()]
```
